### PR TITLE
Nabbed some GC return value fixes from BoH.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Yoinked from https://github.com/BoHBranch/BoH-Bay/pull/530.

Also moved some logic to `/turf/simulated/Destroy()` from `ChangeTurf()`.

### CODE REVIEW

File: code/controllers/subsystem.dm [hunk 1]
```diff
@@ -78,7 +78,7 @@
 	can_fire = 0
 	flags |= SS_NO_FIRE
 	Master.subsystems -= src
-	return ..()
+	. = ..()
```
Comment (afterthought2) [inline code/controllers/subsystem.dm]: ??

File: code/game/objects/items/weapons/tanks/jetpack.dm [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 /obj/item/tank/jetpack/Destroy()
 	qdel(ion_trail)
```
Comment (afterthought2) [inline code/game/objects/items/weapons/tanks/jetpack.dm]: `QDEL_NULL`

File: code/modules/power/breaker_box.dm [hunk 3]
```diff
@@ -20,9 +20,9 @@
 	var/update_locked = 0
 
 /obj/machinery/power/breakerbox/Destroy()
-	..()
 	for(var/datum/nano_module/rcon/R in world)
 		R.FindDevices()
```
Comment (afterthought2) [inline code/modules/power/breaker_box.dm]: Please just kill this. It autoupdates and rebuilds the list on autoupdate anyway.

**Review by afterthought2 — COMMENTED** [review]
Thought we added a unit test for this?

File: code/controllers/subsystem.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/controllers/subsystem.dm]: ???
It's functionally the same, just more consistent.

Comment (afterthought2) [reply code/controllers/subsystem.dm]: I wasn't sure if you were making a stylistic change or if you wanted them to return a different GC hint but didn't include the full change. I'm not sure what GC hint is appropriate here.

File: code/modules/power/breaker_box.dm (hunk 3 above)
Comment (MistakeNot4892) [reply code/modules/power/breaker_box.dm]: Any reason not to kill the SMES one as well?

Comment (afterthought2) [reply code/modules/power/breaker_box.dm]: Kill that too, I feel.

**Comment by MistakeNot4892** [comment]
I don't recall a unit test, but atoms not returning a proper GC hint should cause a stack trace at runtime. I guess maybe none of these items are mapped, or have been blown up recently.

File: code/controllers/subsystem.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/controllers/subsystem.dm]: The PR I am stealing from returns `QDEL_HARRDEL_NOW`. Was going to ask @Lohikar but Discord is crapping out at the moment.

**Comment by afterthought2** [comment]
I was referring to #51

**Review by afterthought2 — APPROVED** [review]

**Comment by PsiOmegaDelta** [comment]
#51 just enforces that `..()` is called, not returned.
The linter can enforce return values but, AFAIK, only that they adhere to a specific type (which would exclude simple numbers)

File: code/controllers/subsystem.dm [hunk 4]
```diff
@@ -78,7 +78,7 @@
 	can_fire = 0
 	flags |= SS_NO_FIRE
 	Master.subsystems -= src
-	return ..()
+	. = QDEL_HINT_HARDDEL_NOW
```
Comment (Lohikar) [inline code/controllers/subsystem.dm]:
```suggestion
	..()
	return QDEL_HINT_HARDDEL_NOW
```

**Review by Lohikar — CHANGES_REQUESTED** [review]

**Comment by afterthought2** [comment]
May be worth changing the GC hint defines so as to function with the linter's typecheck, then, as the actual define values are arbitrary. They have to be datum instances?

**Comment by PsiOmegaDelta** [comment]
Looks like it's possible to enforce specific paths, and not having to resort to instances.
In either case, should be easy to map the existing macros to either.

**Comment by MistakeNot4892** [comment]
I am fine with this PR as-is, changing how GC works is out of scope for what was meant to be a quick fix PR.

File: code/game/turfs/simulated.dm [hunk 5]
```diff
@@ -165,3 +165,12 @@
 	if(GAME_STATE >= RUNLEVEL_GAME)
 		fluid_update()
 	. = ..()
+
+/turf/simulated/Destroy()
```
Comment (Lohikar) [inline code/game/turfs/simulated.dm:169]: This isn't a GC return value fix. Document if intentional, pls.

Comment (MistakeNot4892) [reply code/game/turfs/simulated.dm:169]: ok, it's in the PR message now.

**Review by Lohikar — APPROVED** [review]

**Pull request merged**